src/Authentication/ManagedCredentialsAuthenticator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Keboola\AzureKeyVaultClient\Authentication;
+
+use GuzzleHttp\Exception\GuzzleException;
+use Keboola\AzureKeyVaultClient\Exception\ClientException;
+use Keboola\AzureKeyVaultClient\Exception\InvalidResponseException;
+use Keboola\AzureKeyVaultClient\GuzzleClientFactory;
+use Psr\Log\LoggerInterface;
+
+class ManagedCredentialsAuthenticator implements AuthenticatorInterface
+{
+    /** @var GuzzleClientFactory */
+    private $clientFactory;
+    /** @var LoggerInterface */
+    private $logger;
+
+    const INSTANCE_METADATA_SERVICE_ENDPOINT = 'http://169.254.169.254/';
```

Comment:
To tu ma byt takto zahardkodovane? Aha je to takto dostupne z VM, jasne to je "managed" autorizacia, tak to je asi okej takto
https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service

Reply:
jj, je to "well-known non-routable IP address"